src/pytest_inmanta_lsm/plugin.py

```diff
@@ -148,15 +155,57 @@ def remote_orchestrator_settings() -> Dict[str, Union[str, int, bool]]:
     return {}
 
 
+@pytest.fixture(scope="session")
+def remote_orchestrator_partial(request: pytest.FixtureRequest) -> Iterator[bool]:
+    yield inm_lsm_partial_compile.resolve(request.config)
+
+
+@pytest.fixture(scope="session")
+def remote_orchestrator_project_shared(request: pytest.FixtureRequest, project_shared: Project) -> Iterator[None]:
+    """
+    Shared project to be used by the remote orchestrator. Ensures the module being tested is synced to the remote orchestrator
+    even if it is a v2 module.
+    """
+    in_place = inm_mod_in_place.resolve(request.config)
+    # no need to do anything if this version of inmanta does not support v2 modules or if in_place already adds it to the path
+    if hasattr(module, "ModuleV2") and not in_place:
+        mod: module.Module
+        path: str
+        mod, path = pytest_inmanta.plugin.get_module()
+        # check if a v2 module is installed (no need to do anything for v1 module)
+        installed: Optional[module.ModuleV2] = module.ModuleV2Source(urls=[]).get_installed_module(None, mod.name)
+        if installed is not None:
+            if not installed.is_editable() or not os.path.samefile(mod.path, installed.path):
+                LOGGER.warning(
+                    "The module being tested is not installed in editable mode. To ensure the remote orchestrator uses the same"
+                    " code as the local project, please install the module with `inmanta module install -e .` before running"
+                    " the tests."
+                )
+            # can't rsync and install a non-editable Python package the same way as an editable one (no pyproject.toml/setup.py)
+            # => always use the test dir, even if the module is installed in non-editable mode (the user has been warned)
+            shutil.copytree(mod.path, os.path.join(project_shared._test_project_dir, "libs", mod.name))
```

Comment:
I don't think that is a realistic scenario: the test project directory is a tempdir created specifically for the `project_shared` fixture. Since this is more or less an implementation detail it might still be a good idea to raise an exception or add an assertion, but I don't think it really needs to be targeted at the end user: if it fails it indicates a core invariant of this fixture was invalid.